OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChildServiceTests.cs

```diff
@@ -1,178 +1,286 @@
-//        };
-//    }
-
-//    private IEnumerable<Child> GetTestChildEntities()
-//    {
-//        return new List<Child>()
-//        {
-//            new Child { Id = 1, FirstName = "fn1", LastName = "ln1", MiddleName = "mn1", DateOfBirth = new DateTime(2003, 11, 9), Gender = Gender.Male, ParentId = 1, SocialGroupId = 2 },
-//            new Child { Id = 2, FirstName = "fn2", LastName = "ln2", MiddleName = "mn2", DateOfBirth = new DateTime(2004, 11, 8), Gender = Gender.Female, ParentId = 2, SocialGroupId = 1 },
-//            new Child { Id = 3, FirstName = "fn3", LastName = "ln3", MiddleName = "mn3", DateOfBirth = new DateTime(2006, 11, 2), Gender = Gender.Male, ParentId = 1, SocialGroupId = 1 },
-//        };
-//    }
-//}
\ No newline at end of file
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+using AutoMapper;
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using Moq;
+
+using NUnit.Framework;
+
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.Tests.Common;
+using OutOfSchool.WebApi.Config;
+using OutOfSchool.WebApi.Models;
+using OutOfSchool.WebApi.Services;
+
+namespace OutOfSchool.WebApi.Tests.Services;
+
+[TestFixture]
+public class ChildServiceTests
+{
+    private Mock<IParentRepository> parentRepositoryMock;
+    private Mock<IEntityRepositorySoftDeleted<Guid, Child>> childRepositoryMock;
+    private Mock<IEntityRepositorySoftDeleted<long, SocialGroup>> socialGroupRepositoryMock;
+    private Mock<ILogger<ChildService>> loggerMock;
+    private Mock<IMapper> mapperMock;
+    private Mock<IApplicationRepository> applicationRepositoryMock;
+    private Mock<IOptions<ParentConfig>> parentConfigMock;
+    private ChildService childService;
+
+    private OffsetFilter offsetFilter = new OffsetFilter();
+
+
+    [SetUp]
+    public void SetUp()
+    {
+        parentRepositoryMock = new Mock<IParentRepository>();
+        childRepositoryMock = new Mock<IEntityRepositorySoftDeleted<Guid, Child>>();
+        socialGroupRepositoryMock = new Mock<IEntityRepositorySoftDeleted<long, SocialGroup>>();
+        loggerMock = new Mock<ILogger<ChildService>>();
+        mapperMock = new Mock<IMapper>();
+        applicationRepositoryMock = new Mock<IApplicationRepository>();
+        parentConfigMock = new Mock<IOptions<ParentConfig>>();
+
+        childService = new ChildService(
+            childRepositoryMock.Object,
+            parentRepositoryMock.Object,
+            socialGroupRepositoryMock.Object,
+            loggerMock.Object,
+            mapperMock.Object,
+            applicationRepositoryMock.Object,
+            parentConfigMock.Object);
+    }
+
+    [Test]
+    public async Task GetApprovedByWorkshopId_ExistingChildren_ShouldReturnChildren()
```

Comment:
(Test logic) It looks that this test would never be "red", as it checks something, but not you code changes. 

Try to remove from your `ChildService` added piece of code `|| p.Status == ApplicationStatus.StudyingForYears` and this test will not fail.

Reply:
Hopefully, I have fixed the tests, and now they will fail when the filter in ChildService is changed